Fixes from Clang compilation errors

PetrilloAtWork opened the pull request:
Clang 14 (`c14:prof`) complains about a few lines of code.
While the merit of the complains is arguable (not plain wrong, nor right), "fixing" the code is cheap, and needed if we want to compile with Clang.
The fixes are simple and technical.

Reviewers: who is the Collaboration C++ expert?

SFBayLaser commented:
If you translate from C++ to FORTRAN II then I will be happy to review.

mmrosenberg commented:
trigger build SBNSoftware/icarusalg#87

FNALbuild commented:
:heavy_check_mark: CI build for LArSoft Succeeded on slf7 for c14:prof -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app//view_builds/index?offset=0&builds=lar_ci/23302&builds=)

FNALbuild commented:
:warning: CI build for ICARUS Warning at phase ci_tests ICARUS on slf7 for e26:prof - ignored warnings for build -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/ns:ICARUS/view_builds/index?offset=0&builds=icarus_ci/10335&builds=)

:rotating_light: For more details about the warning phase, check the [ci_tests ICARUS phase logs](https://dbweb0.fnal.gov/LarCI/app/ns:ICARUS/build_detail/phase_details?build_id=icarus_ci/10335&platform=Linux%20slf7&phase=ci_tests&buildtype=slf7%20e26:prof)

parent CI build details are available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/23301&builds=)

FNALbuild commented:
:x: CI build for ICARUS Failed at phase ci_tests ICARUS on slf7 for c14:prof - ignored failure for unit_test - ignored warnings for build -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/ns:ICARUS/view_builds/index?offset=0&builds=icarus_ci/10334&builds=)

:rotating_light: For more details about the failed phase, check the [ci_tests ICARUS phase logs](https://dbweb0.fnal.gov/LarCI/app/ns:ICARUS/build_detail/phase_details?build_id=icarus_ci/10334&platform=Linux%20slf7&phase=ci_tests&buildtype=slf7%20c14:prof)

parent CI build details are available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/23302&builds=)

PetrilloAtWork commented:
Forced push to update this branch to `master` (`develop` is a bit too ahead and is currently failing compilation for other reasons).
I have added fixes from a few new warnings from Clang.
I have also fixed what I think is a *bug* in the code, which made the relevant unit test fail above but only under Clang (I am not exactly sure why, and some undefined behaviour is certainly involved).

FNALbuild commented:
:heavy_check_mark: CI build for LArSoft Succeeded on slf7 for c14:prof -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app//view_builds/index?offset=0&builds=lar_ci/23963&builds=)

FNALbuild commented:
:heavy_check_mark: CI build for LArSoft Succeeded on slf7 for e26:prof -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app//view_builds/index?offset=0&builds=lar_ci/23962&builds=)

FNALbuild commented:
:x: CI build for ICARUS Failed at phase ci_tests ICARUS on slf7 for c14:prof - ignored warnings for build -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/ns:ICARUS/view_builds/index?offset=0&builds=icarus_ci/10795&builds=)

:rotating_light: For more details about the failed phase, check the [ci_tests ICARUS phase logs](https://dbweb0.fnal.gov/LarCI/app/ns:ICARUS/build_detail/phase_details?build_id=icarus_ci/10795&platform=Linux%20slf7&phase=ci_tests&buildtype=slf7%20c14:prof)

parent CI build details are available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/23963&builds=)

FNALbuild commented:
:warning: CI build for ICARUS Warning at phase ci_tests ICARUS on slf7 for e26:prof - ignored warnings for build -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/ns:ICARUS/view_builds/index?offset=0&builds=icarus_ci/10796&builds=)

:rotating_light: For more details about the warning phase, check the [ci_tests ICARUS phase logs](https://dbweb0.fnal.gov/LarCI/app/ns:ICARUS/build_detail/phase_details?build_id=icarus_ci/10796&platform=Linux%20slf7&phase=ci_tests&buildtype=slf7%20e26:prof)

parent CI build details are available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/23962&builds=)
